### ISSUE TITLE
Distinction between "original" and derived values

### ISSUE THREAD

**Issue by ghost** [enanomapper/data.enanomapper.net#15]
It would be great to have a clear distinction between "original" and derived features together with a method to retrieve the function for calculating derived values. I will take the protein corona data as an example: It has a couple of "Change" features that indicate the difference of certain values in human serum. In an user friendly search form I do not want to ask for "zeta potential", "zeta potential (human serum)" _and_ "zeta potential change", because the difference (change) can be calculated easily from the other 2 parameters. At the moment I have to use a handcrafted solution, but it would be great to retrieve the equation "zeta potential change" = "zeta potential (human serum)" - "zeta potential" directly from the database. A similar mechanism could be useful to calculate means, stdevs, etc or maybe even for dose-response analysis. And for tox endpoints, it is unnecessary to make two predictions for cell association
 and the log2 value, if both values are related.
 I am however not sure how to implement it technically (maybe via feature ontologies, but how to represent functions in a language agnostic form?) and if it would be worth the efforts.

**Comment by vedina** [comment]
Valid question, there is a related concept in BAO, which support derived values (like IC50 is derived from the dose response values). We have been thinking of implementing it by storing both types of values (when available), but providing also the link between the two (and eventually the type of relation, formula, protocol, etc.). Also in ISA-TAB this is represented by providing a link to the "data transformation" protocol. Hence one may have the statement  `value X` is processed by (computational) `protocol Y` and  the result is `Value Z`

For this particular case may be it was not wise to import everything form the original Protein corona file, but be selective. 

The search is a different story (related to #8)

**Comment by ghost** [comment]
> Also in ISA-TAB this is represented by providing a link to the "data transformation" protocol. Hence one may have the statement value X is processed by (computational) protocol Y and the result is Value Z.

Yes I was thinking along these lines, but with a machine readable and language agnostic (computational) `protocol Y` (maybe I am just dreaming). If there is a clear distinction and relation between features I think it does not matter if derived values are stored or not (may depend on other factors like storage space vs computation time, access by search engines etc).

**Comment by vedina** [comment]
it's close to a dream :) besides the cases of a simple formula.
